mathml-block.php

```diff
@@ -28,45 +80,127 @@ function mathml_block_enqueue_scripts() {
+/**
+ * Register block.
+ */
+function register_block() {
+	if ( ! function_exists( 'register_block_type' ) ) {
+		return;
+	}
 
-	// Filter the MathJax config string.
-	$config_string = apply_filters( 'mathml_block_mathjax_config', 'TeX-MML-AM_CHTML' );
+	$registry = WP_Block_Type_Registry::get_instance();
 
-	wp_enqueue_script(
-		'mathjax',
-		'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/MathJax.js?config=' . $config_string
+	// @todo This can probably be de-duplicated in the JS code with registerBlockType.
+	$attributes = array(
+		'formula' => array(
+			'source'   => 'html',
+			'selector' => 'div',
+			'type'     => 'string',
+		),
 	);
 
+	if ( $registry->is_registered( BLOCK_NAME ) ) {
+		$block                  = $registry->get_registered( BLOCK_NAME );
+		$block->render_callback = __NAMESPACE__ . '\render_block';
+		$block->attributes      = array_merge( $block->attributes, $attributes );
+	} else {
+		register_block_type(
+			BLOCK_NAME,
+			[
+				'render_callback' => __NAMESPACE__ . '\render_block',
+				'attributes'      => $attributes,
+			]
+		);
+	}
 }
-add_action( 'enqueue_block_editor_assets', __NAMESPACE__ . '\mathml_block_enqueue_scripts' );
+add_action( 'init', __NAMESPACE__ . '\register_block' );
 
 /**
- * Potentially enqueue the front end mathjax script, if any mathml blocks are detected in the content.
+ * Add async attribute to MathJax script tag.
+ *
+ * @param string $tag    Script tag.
+ * @param string $handle Script handle.
+ *
+ * @return string Script tag.
  */
-function potentially_add_front_end_mathjax_script() {
-	global $post;
-
-	// Only apply on singular pages.
-	if ( ! is_singular() ) {
-		return;
+function add_async_to_mathjax_script_loader_tag( $tag, $handle ) {
+	if ( MATHJAX_SCRIPT_HANDLE === $handle ) {
+		$tag = preg_replace( '/(?<=<script\s)/', ' async ', $tag );
 	}
+	return $tag;
+}
 
-	// Check the content for mathml blocks.
-	$has_mathml_block  = strpos( $post->post_content, 'wp:mathml/mathmlblock' );
-	$has_mathml_inline = strpos( $post->post_content, '<mathml>' );
-	if ( false === $has_mathml_block && false === $has_mathml_inline ) {
-		return;
+/**
+ * Render block.
+ *
+ * Creates an <amp-mathml> element on AMP responses.
+ *
+ * @param array  $attributes Attributes.
+ * @param string $content    Content.
+ *
+ * @return string Rendered block.
+ */
+function render_block( $attributes, $content = '' ) {
+	if ( is_admin() || ! preg_match( '#^(?P<start_div>\s*<div.*?>)(?P<formula>.+)(?P<end_div></div>\s*)$#s', $content, $matches ) ) {
+		return $content;
 	}
 
-	// Filter the MathJax config string.
-	$config_string = apply_filters( 'mathml_block_mathjax_config', 'TeX-MML-AM_CHTML' );
+	if ( is_amp() ) {
+		static $printed_style = false;
+		if ( ! $printed_style ) {
+			// Add same margins as .MJXc-display.
+			?>
+			<style class="amp-mathml">
+				.wp-block-mathml-mathmlblock amp-mathml { margin: 1em 0; }
+			</style>
+			<?php
```

Comment:
This was wrong. It caused a bug in legacy Reader mode. See https://github.com/adamsilverstein/mathml-block/pull/38.